turtle_tf2_cpp/src/turtle_tf2_listener.cpp

```diff
@@ -88,38 +67,87 @@ class FrameListener : public rclcpp::Node
+          );
+          continue;
+        }
+        RCLCPP_INFO(
+          this->get_logger(),
+          "Service not available, waiting again..."
+        );
+      }
+
+      // Initialize request with turtle name and coordinates
+      // Note that x, y and theta are defined as floats in turtlesim/srv/Spawn
+      auto request = std::make_shared<turtlesim::srv::Spawn::Request>();
+      request->x = 4.0;
+      request->y = 2.0;
+      request->theta = 0.0;
+      request->name = "turtle2";
+      // Call request
+      auto result = spawner_->async_send_request(request);
+
+      turtle_spawned_ = true;
     }
 
-    geometry_msgs::msg::Twist msg;
+    if (!transform_available_) {
+      const auto timeout_ms_ = 100ms;
+      while (!tf_buffer_->canTransform(toFrameRel, fromFrameRel, tf2::TimePointZero, 0ms) &&
+        rclcpp::ok())
+      {
+        RCLCPP_INFO(
+          get_logger(), "waiting %ld ms for %s->%s transform to become available",
+          timeout_ms_.count(), toFrameRel.c_str(), fromFrameRel.c_str());
+        std::this_thread::sleep_for(timeout_ms_);
+      }
+      RCLCPP_INFO(get_logger(), "transform available");
+      transform_available_ = true;
+    }
+
+    if (transform_available_) {
+      geometry_msgs::msg::TransformStamped transformStamped;
+
+      // Look up for the transformation between target_frame and turtle2 frames
+      // and send velocity commands for turtle2 to reach target_frame
+      try {
+        transformStamped = tf_buffer_->lookupTransform(
+          toFrameRel, fromFrameRel,
+          tf2::TimePointZero,
+          0ms);
+      } catch (tf2::LookupException & ex) {
+        RCLCPP_INFO(this->get_logger(), "transform not ready");
+        return;
+      }
 
-    static const double scaleRotationRate = 1.0;
-    msg.angular.z = scaleRotationRate * atan2(
-      transformStamped.transform.translation.y,
-      transformStamped.transform.translation.x);
+      geometry_msgs::msg::Twist msg;
 
-    static const double scaleForwardSpeed = 0.5;
-    msg.linear.x = scaleForwardSpeed * sqrt(
-      pow(transformStamped.transform.translation.x, 2) +
-      pow(transformStamped.transform.translation.y, 2));
+      static const double scaleRotationRate = 1.0;
+      msg.angular.z = scaleRotationRate * atan2(
+        transformStamped.transform.translation.y,
+        transformStamped.transform.translation.x);
 
-    publisher_->publish(msg);
+      static const double scaleForwardSpeed = 0.5;
+      msg.linear.x = scaleForwardSpeed * sqrt(
+        pow(transformStamped.transform.translation.x, 2) +
+        pow(transformStamped.transform.translation.y, 2));
+
+      publisher_->publish(msg);
+    }
   }
+  bool turtle_spawned_;
+  bool transform_available_;
+  rclcpp::Client<turtlesim::srv::Spawn>::SharedPtr spawner_;
   rclcpp::TimerBase::SharedPtr timer_;
+  rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr publisher_;
   std::shared_ptr<tf2_ros::TransformListener> transform_listener_;
```

Comment:
```suggestion
  rclcpp::Client<turtlesim::srv::Spawn>::SharedPtr spawner_{nullptr};
  rclcpp::TimerBase::SharedPtr timer_{nullptr};
  rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr publisher_{nullptr};
  std::shared_ptr<tf2_ros::TransformListener> transform_listener_{nullptr};
```

Reply:
Updated in https://github.com/ros/geometry_tutorials/pull/53/commits/5b44b06d2e1d53678c7c602ca5d7d36103c70dae